staging/maven-plugin/src/main/java/org/sonatype/nexus/maven/staging/deploy/strategy/StagingDeployStrategy.java

```diff
@@ -88,6 +98,55 @@ public void deployPerModule(final DeployPerModuleRequest request)
   }
 
   /**
+   * Store nexusUrl in a separate file. It will be used when deploying to remote remote staging repository.
+   *
+   * @param request
+   * @param stagingDirectory
+   * @throws MojoExecutionException
+   */
+  private void storeNexusUrl(DeployPerModuleRequest request, File stagingDirectory) throws MojoExecutionException
+  {
+    FileOutputStream fos = null;
+    try{
+      final File nexusUrlFile = new File(stagingDirectory, ".nexusUrl");
+      if(nexusUrlFile.exists()) {
+        return;
+      }
+      fos = new FileOutputStream(nexusUrlFile);
+      fos.write(request.getParameters().getNexusUrl().getBytes("ISO-8859-1"));
```

Comment:
this write happens with ISO-8859-1 but the read is not explicit about encoding, I would propose to have both ends read/write UTF-8. At the same time there should be a method in plexus-utils or google-commons (if the project using g-c) that reads the content of a file into String